source/intro.md

```diff
@@ -417,131 +418,195 @@ Wait a minute, nothing happened this time! Where's our data?
 Actually, something did happen: the data was loaded in 
 and now has the name `can_lang` associated with it. 
 And we can use that name to access the data frame and do things with it. 
-For example, we can type the name of the data frame to print the first few rows 
-on the screen. You will also see at the top that the number of observations (i.e., rows) and 
-variables (i.e., columns) are printed. Printing the first few rows of a data frame 
-like this is a handy way to get a quick sense for what is contained in a data frame.
+For example, we can type the name of the data frame to print both the first few rows 
+and the last few rows, separated by `...`,
+on the screen. You will also see at the bottom that the number of observations (i.e., rows) and 
+variables (i.e., columns) are printed. Printing a summary of a data frame 
+like this is a handy way to get a quick sense for what is contained in it.
 
 ```{code-cell} ipython3
 can_lang
 ```
 
-## Creating subsets of data frames with `df[]` & `loc[]`
+## Creating subsets of data frames with `[]` & `loc[]`
 
-```{index} pandas.DataFrame; df[], pandas.DataFrame; loc[]
+```{index} pandas.DataFrame; [], pandas.DataFrame; loc[]
 ```
 
 Now that we've loaded our data into Python, we can start wrangling the data to
 find the ten Aboriginal languages that were most often reported
-in 2016 as mother tongues in Canada. In particular, we will construct 
+in 2016 as mother tongues in Canada. In particular, we want to construct 
 a table with the ten Aboriginal languages that have the largest 
-counts in the `mother_tongue` column. 
-The `df[]` and `loc[]` properties of the `pandas` dataframe will help us
-here. The `df[]` property allows you to obtain a subset of the
-rows with specific values, while the `loc[]` property allows you 
-to obtain a subset of the columns. Therefore, we can use `df[]` 
-to filter the rows to extract the Aboriginal languages in the data set, and 
-then use `loc[]` property to obtain only the columns we want to include in our table.
-
-### Using `df[]` to extract rows
+counts in the `mother_tongue` column. The first step is to extract
+from our `can_lang` data only those rows that correspond to Aboriginal languages,
+and then the second step is to keep only the `language` and `mother_tongue` columns.
+The `[]` and `loc[]` operations on the `pandas` data frame will help us
+here. The `loc[]` operation allows you to obtain a subset of *both* the rows
+and columns of a data frame at once. In order to tell the `loc[]` operation
+which subset of columns we want to obtain, we can just list their names.
+In order to tell the `loc[]` operation which subset of rows we want to obtain, 
+we can examine the values of a single column of the data frame using the `[]` operation. 
+In our example in particular, we will use 
+`[]` to look at the `category` column in `can_lang` to extract only the 
+Aboriginal languages in the data set.
```

Comment:
After thinking about it yesterday, I think I am on board with your idea. I've got two issues with it, one sort of unimportant, and one more important.

Unimportant: I find it aesthetically unappealing that `[ ]` is used for selecting rows, or selecting columns, and then you need a totally separate thing `.loc[ ]` to select both rows and columns. I am happy as long as we resolve this whole mess with `[ ]`, `loc[ ]`, `iloc[ ]`, series, and indices in Chapter 3.

More important: How should we best explain the following
```
can_lang[can_lang["category"] == "Aboriginal languages"]
```
*without* mentioning series?

I can imagine the appearance of `can_lang` twice in that expression being kind of confusing for students. I will do a version of it and see how it goes...

Reply:
(By the way -- I am going to keep the original `filter` then `select` ordering from the R textbook, since this is fine as-is. If I wanted to write this from scratch I would indeed swap them, but I don't have much time left for this and am prioritizing important things.)

Reply:
The only thing I think could be confusing is that inside that boolean expression is a column selection with `[]` which we don't explain until the next section (`can_lang['category'] == 'Aboriginal languages'`), which I think we could solve by swapping the order of the section, but I saw your note about that you didn't want to prioritize that now which I also think makes sense.